### PR TITLE
[clang-format] Rename option AlwaysBreakAfterReturnType.

### PR DESCRIPTION
**Pull request by rmarker**
Changes the option to BreakAfterReturnType option, with a more relevant name, deprecating and replacing AlwaysBreakAfterReturnType.
Following up on #78010.

### CODE REVIEW

**Comment by rmarker** [comment]
@owenca, here is the PR with `BreakAfterReturnType`.
I think I got everything necessary to deprecate and replace `AlwaysBreakAfterReturnType`.

File: clang/include/clang/Format/Format.h [hunk 1]
```diff
@@ -1010,7 +1010,8 @@ struct FormatStyle {
   /// \version 3.7
   DefinitionReturnTypeBreakingStyle AlwaysBreakAfterDefinitionReturnType;
 
-  /// The function declaration return type breaking style to use.
+  /// The function declaration return type breaking style to use.  This
+  /// option is **deprecated** and is retained for backwards compatibility.
   /// \version 3.8
   ReturnTypeBreakingStyle AlwaysBreakAfterReturnType;
```
Comment (HazardyKnusperkeks) [inline clang/include/clang/Format/Format.h:1016]: The actual member has to go.

**Comment by owenca** [comment]
@rmarker we only need to change the user-facing part and should leave the internal naming unchanged (at least in this patch) in order to have the smallest diff possible. I've added the support for deprecating/renaming top-level struct-type options to `dump_format_style.py` in #81093.

File: clang/include/clang/Format/Format.h (hunk 1 above)
Comment (owenca) [reply clang/include/clang/Format/Format.h:1016]: +1, but see https://github.com/llvm/llvm-project/pull/80827#issuecomment-1935386091.

**Comment by owenca** [comment]
See the follow-up commit 7664ddf88112 to #81093.

**Comment by rmarker** [comment]
Right. I've updated the PR, following the pattern of #81093.

**Review by owenca — APPROVED** [review]

File: clang/unittests/Format/ConfigParseTest.cpp [hunk 2]
```diff
@@ -677,6 +677,23 @@ TEST(ConfigParseTest, ParsesConfiguration) {
...
+              FormatStyle::RTBS_All);
+  CHECK_PARSE("BreakAfterReturnType: TopLevel", AlwaysBreakAfterReturnType,
+              FormatStyle::RTBS_TopLevel);
+  CHECK_PARSE("BreakAfterReturnType: AllDefinitions",
+              AlwaysBreakAfterReturnType, FormatStyle::RTBS_AllDefinitions);
+  CHECK_PARSE("BreakAfterReturnType: TopLevelDefinitions",
+              AlwaysBreakAfterReturnType,
+              FormatStyle::RTBS_TopLevelDefinitions);
+  // For backward compatibility:
   Style.AlwaysBreakAfterReturnType = FormatStyle::RTBS_All;
```
Comment (owenca) [inline clang/unittests/Format/ConfigParseTest.cpp:697]: Redundant.

**Comment by rmarker** [comment]
Thanks for updating `ConfigParseTest.cpp` @owenca and merging.

Do you also want to do the subsequent NFC commit directly? Like [7664ddf](https://github.com/llvm/llvm-project/commit/7664ddf8811242295abb837640cad8dd8cefb5e8).
Or do you want me to make another pull request for that?

**Comment by owenca** [comment]
Feel free to open another PR if you're interested. 🙂

**Pull request merged**